src/screens/transaction-details/components/TransactionMasthead.tsx

```diff
@@ -289,8 +314,28 @@ export default function TransactionMasthead({ transaction }: { transaction: Rain
   }, []);
 
   const outputAsset = useMemo(() => {
-    const outAsset = transaction?.changes?.find(a => a?.direction === 'out')?.asset;
-    if (!outAsset) return undefined;
+    const change = transaction?.changes?.find(a => a?.direction === 'out');
+
+    if (!change?.asset) return undefined;
+
+    // NOTE: For pending transactions let's use the change value
+    // since the balance hasn't been updated yet.
+    if (['swap', 'wrap', 'unwrap'].includes(transaction.type) && transaction.status === 'pending') {
```

Comment:
i've seen this condition a few times, could be good to extract it out into a helper called isPendingSwap or something